python/SECRETSMANAGER_MAX_SECRET_AGE.py

```diff
@@ -0,0 +1,77 @@
+#
+# This file made available under CC0 1.0 Universal (https://creativecommons.org/publicdomain/zero/1.0/legalcode)
+#
+# Ensure that Secret LastChangedDate is no older than days specified
+# Description: Checks that Secrets have been rotated since number of days
+#
+# Trigger Type: Periodic
+# Scope of Changes: N/A
+# Accepted Parameters: max_secret_age_days
+# Example Values: 1, 2, 3,... default 30
+#
+# requires boto3 >= 1.9.152, botocore >= 1.12.152
+# will fail without these library versions as pagination
+# was not supported with the below libraries included
+# in Lambda as of May 21st 2019
+# Python 3.7 Lambda includes boto3 "1.9.42"
+# Python 3.6 Lambda includes boto3 "1.7.74"
+# You must use a custom deployment package that includes
+# the required boto3 and botocore versions
+# https://aws.amazon.com/premiumsupport/knowledge-center/build-python-lambda-deployment-package/
+
+import json
+from datetime import datetime, timedelta, timezone
+import boto3
+
+def evaluate_compliance(rule_parameters, secret):
+    if 'max_secret_age_days' in rule_parameters:
+        max_secret_age = datetime.now(timezone.utc) - timedelta(days=int(rule_parameters['max_secret_age_days']))
+    else:
+        max_secret_age = datetime.now(timezone.utc) - timedelta(days=30)
+
+    if 'LastRotatedDate' in secret and datetime.replace(secret['LastRotatedDate'], tzinfo=timezone.utc) > max_secret_age:
+        return "COMPLIANT"
+    if datetime.replace(secret['LastChangedDate'], tzinfo=timezone.utc) > max_secret_age:
+        return "COMPLIANT"
+    return "NON_COMPLIANT"
+
+def lambda_handler(event, context):
+    if boto3.__version__ < '1.9.152':
+        print('boto3 version too old for secret pagination')
+        print('boto3 version: ' + boto3.__version__)
+        exit(1)
+
+    now = datetime.now(timezone.utc)
+    if 'ruleParameters' in event:
+        rule_parameters = json.loads(event['ruleParameters'])
+    else:
+        print('No rule paramters found')
```

Comment:
You have a default value of 30, so it should be optional, not throwing an error.

Reply:
I'm confused about this?? It only throws an error if you were to run it directly in lambda as Config always provides rule parameters in the event even if you don't specify any parameters. This should only throw an error when not run by ConfigService.

Reply:
The idea is that if the customer does not provide any input parameters, then the rule should default to 30. The rule (https://docs.aws.amazon.com/config/latest/developerguide/acm-certificate-expiration-check.html) works this way, if you put no parameter, it defaults to 14 days.

Reply:
I understand about defaulting to 30 for a value within rule_parameters, but that has nothing to do with this portion of code. This portion of code will never raise an exception when run by a Config Rule.

Reply:
L49 as an `exit(1)`

Reply:
This is what I'm saying though, L49 makes rule_parameters mandatory, not rule_parameters['max_secret_age_days'], which is optional.

Reply:
I removed this entirely and as the RDK code already has code for this.